test/checkpoint.test.js

```diff
@@ -1,28 +1,43 @@
 var async = require('async');
 var loopback = require('../');
-
-// create a unique Checkpoint model
-var Checkpoint = loopback.Checkpoint.extend('TestCheckpoint');
-
-var memory = loopback.createDataSource({
-  connector: loopback.Memory
-});
-Checkpoint.attachTo(memory);
+var expect = require('chai').expect;
 
 describe('Checkpoint', function() {
-  describe('current()', function() {
+  var Checkpoint;
+  describe('bump and current', function() {
+    beforeEach(function() {
+      // create a unique Checkpoint model
+      Checkpoint = loopback.Checkpoint.extend('TestCheckpoint');
```

Comment:
Hmm, this does not work exactly the way you think it works. There is only one global model registry, this code is rewriting the entry under the key `TestCheckpoint`. If you really want unique model, then you have to use an unique model name.

However, I think in this particular case, it's ok to share the same model, just attach it to a unique data source in each test.

```js
var Checkpoint = loopback.Checkpoint.extend('TestCheckpoint');
//...
beforeEach(function() {
  var memory = loopback.createDataSource(/*...*/);
  Checkpoint.attachTo(memory);
});
```